test/e2e/specs/azure_monitor.go

```diff
@@ -0,0 +1,142 @@
+package specs
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/operationalinsights/mgmt/2015-11-01-preview/operationalinsights"
+	azresources "github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
+	"github.com/Azure/go-autorest/autorest/to"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/openshift-azure/pkg/util/random"
+	"github.com/openshift/openshift-azure/pkg/util/ready"
+	"github.com/openshift/openshift-azure/test/clients/azure"
+	"github.com/openshift/openshift-azure/test/manifests"
+	"github.com/openshift/openshift-azure/test/sanity"
+)
+
+const (
+	azureMonitorProject = "openshift-azure-logging"
+	nodeAgentName       = "log-analytics-node-agent"
+	clusterAgentName    = "log-analytics-cluster-agent"
+)
+
+var _ = Describe("Azure Red Hat OpenShift e2e tests for Azure Monitor Integration [EveryPR]", func() {
+	withAzureMonitorIt("should be possible for a customer to update the azure monitor configuration for their cluster [EndUser]", func() {
+		ctx := context.Background()
+		rgName, err := random.LowerCaseAlphanumericString(5)
+		Expect(err).ToNot(HaveOccurred())
+		rgName = "e2e-workspace-" + rgName
+
+		By("Reading the cluster config before the azure monitor checks")
+		clusterConfig, err := azure.RPClient.OpenShiftManagedClusters.Get(ctx, os.Getenv("RESOURCEGROUP"), os.Getenv("RESOURCEGROUP"))
+		Expect(err).NotTo(HaveOccurred())
+		Expect(clusterConfig).NotTo(BeNil())
+		Expect(*clusterConfig.Properties.MonitorProfile.Enabled).To(Equal(true))
+		originalWorkapceId := *clusterConfig.Properties.MonitorProfile.WorkspaceResourceID
+
+		By(fmt.Sprintf("Creating workspace resource group %s", rgName))
+		workspaceGroup, err := azure.RPClient.Groups.CreateOrUpdate(ctx, rgName, azresources.Group{
+			Location: clusterConfig.Location,
+			Tags: map[string]*string{
+				"now": to.StringPtr(fmt.Sprintf("%d", time.Now().Unix())),
+				"ttl": to.StringPtr("3h"),
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(workspaceGroup).NotTo(BeNil())
+		defer azure.RPClient.Groups.Delete(ctx, rgName)
+
+		By(fmt.Sprintf("Creating workspace %s in resource group %s", rgName, rgName))
```

Comment:
I think we should not be creating workspaces on each e2e run, have pre-made workspaces and just use them..
Anubhuti said these were quite heavy weight resources and we should share them across clusters.

Reply:
I am not sure that using pre-made workspaces would be cheaper. These workspaces created in the e2e live for less than 5 minutes and are deleted at the end. I think it is important to create these workspaces here so we can test the workspace update functionality as well as the container insights solution deployment process for workspaces

Reply:
I think I'm OK with this.